### PR TITLE
D ui3 42 private bool  has subscribed at the least volatile should be added but the pattern is not air tight is this is a problem

### PR DESCRIPTION
**Pull request by adamhathcock**
https://spockle.atlassian.net/browse/DUI3-42

### CODE REVIEW

File: Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/IRevitIdleManager.cs [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-// POC: needs interface
-// is probably misnamed, perhaps OnIdleCallbackManager
-internal interface IRevitIdleManager
-{
-  /// <summary>
-  /// Subscribe deferred action to Idling event to run it whenever Revit becomes idle.
-  /// </summary>
-  /// <param name="action"> Action to call whenever Revit becomes Idle.</param>
-  /// some events in host app are trigerred many times, we might get 10x per object
-  /// Making this more like a deferred action, so we don't update the UI many times
```
Comment (JR-Morgan) [inline Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/IRevitIdleManager.cs:12]: double check if this documentation needs copying over

File: Connectors/Rhino/Speckle.Connectors.Rhino7/Bindings/RhinoSelectionBinding.cs [hunk 2]
```diff
@@ -30,10 +30,11 @@ ITopLevelExceptionHandler topLevelExceptionHandler
...
   }
 
-  void OnSelectionChange(object o, EventArgs eventArgs)
-  {
-    _idleManager.SubscribeToIdle(() => _topLevelExceptionHandler.CatchUnhandled(UpdateSelection));
-  }
+  void OnSelectionChange(object o, EventArgs eventArgs) =>
+    _idleManager.SubscribeToIdle(
+      nameof(RhinoSelectionBinding),
+      () => _topLevelExceptionHandler.CatchUnhandled(UpdateSelection)
```
Comment (JR-Morgan) [inline Connectors/Rhino/Speckle.Connectors.Rhino7/Bindings/RhinoSelectionBinding.cs:36]: If we're doing top level exception handling in the idle manager, we could remove the top level here

File: Connectors/Rhino/Speckle.Connectors.Rhino7/Bindings/RhinoSendBinding.cs [hunk 3]
```diff
@@ -117,26 +117,12 @@ private void SubscribeToRhinoEvents()
...
         ChangedObjectIds.Add(e.NewRhinoObject.Id.ToString());
         ChangedObjectIds.Add(e.OldRhinoObject.Id.ToString());
-        _idleManager.SubscribeToIdle(RunExpirationChecks);
+        _idleManager.SubscribeToIdle(nameof(RhinoSendBinding), RunExpirationChecks);
       });
   }
 
   public List<ISendFilter> GetSendFilters() => _sendFilters;
 
-  public List<CardSetting> GetSendSettings()
```
Comment (JR-Morgan) [inline Connectors/Rhino/Speckle.Connectors.Rhino7/Bindings/RhinoSendBinding.cs:126]: I don't know what this is

File: Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/IRevitIdleManager.cs (hunk 1 above)
Comment (adamhathcock) [reply Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/IRevitIdleManager.cs:12]: this is moved

**Review by JR-Morgan — APPROVED** [review]

**Pull request merged**